daemon/dns/dns_resolvconf_file.go

```diff
@@ -94,3 +101,85 @@ func restoreDNS() error {
 	}
 	return internal.FileWrite(resolvconfFilePath, out, internal.PermUserRWGroupROthersR)
 }
+
+func tryToRestoreDNS() {
+	// if target is symlink, probably it is managed by other software - do nothing
+	if internal.IsSymLink(resolvconfFilePath) {
+		return
+	}
+	// if target /etc/resolv.conf contains nordvpn changes:
+	// if possible restore from backup
+	// if backup does not exists, create simple dns settings file
+	out, err := internal.FileRead(resolvconfFilePath)
+	if err != nil {
+		log.Println(internal.ErrorPrefix, fmt.Errorf("reading resolv.conf: %w", err))
+		return
+	}
+	if !strings.Contains(string(out), resolvconfFileMark) {
+		return
+	}
+
+	log.Println(internal.WarningPrefix, "/etc/resolv.conf contains our changes - need to fix this")
+
+	// try to unlock, if file was locked
+	_ = internal.FileUnlock(resolvconfFilePath)
+	// restore from backup if backup file exists
+	if internal.FileExists(resolvconfBackupPath) {
```

Comment:
Shouldn't there be some checks for the backup file, at least if is not symlink?
Because, at the moment if backup file exists then the app doesn't do anything, but at restore will just replace `/etc/resolv.conf` with its content.
Or change the logic from setDns and always overwrite the backup file, maybe even mark it as immutable?

Reply:
Backup file is root protected - no need for extra protection, now on restore I have added check for backup file not to have our mark, backup will be removed. IMO, that is enough.